### DISCUSSION TITLE
rtsp-simple-server.yml as object to be handled by an external database

### DISCUSSION THREAD

**Discussion by babashunsuke** [bluenviron/mediamtx#428] [Questions]
I setup the rtsp-simple-server on a cloud to publish a stream from my edge inside NAT. I wrote the code with gstreamer and it works fine. As I did not want to use a common user & password for streams, I modified the rtsp-simple-server to use credential for basic authentication in an external redis server. I can use different password for each stream path and it can be changed dynamically. 

If the contents of rtsp-simple-server.yml is handled by an external database, setting will become more flexible and rtsp-simple-server will become more useful. Otherwise, I want to configure read credential and publish credential for each stream path by an external database. I think just starting from credential will be easy. 

I want your opinion.
1. Is there a demand for such a function for you?
2. what is the best approach to realize this in terms of coding point of view ?

**Comment by aler9** [comment]
Hello, this can be done with an API, that is currently under development; the dedicated issue is this:

https://github.com/aler9/rtsp-simple-server/issues/156

[a comment by an automated account is not shown]
